rules/xcodeproj.bzl

```diff
@@ -1142,6 +1167,49 @@ Tags for configuration:
 
         If not present the 'Debug' and 'Release' Xcode build configurations will be created by default without
         appending any additional bazel invocation flags.
+        """),
+        "build_type_by_config": attr.string_dict(default = {}, doc = """
```

Comment:
One of the reasons behind the original config design API we wanted to pass an array of configs to Bazel was to prevent the xcodeproj from diverging from CLI. Shouldn't these be already available to the build system - specifically that Bazel will be setting these attributes?

The aspect would read in these values from `ios_application.bundle_id` for instance

Reply:
The value is available when the aspect is collecting info but that's only for the config active while the proj is being generated, meaning that one has to generate multiple projects one in each config flavour. 

With the setup I'm proposing here one can map configs to `debug/release` and set any flavour specific setting in `target_settings_by_config` (e.g. `PRODUCT_BUNDLE_IDENTIFIER` is relevant when deploying to device) so that building things on that config is just a matter of selecting a different Xcode build configuration on the GUI instead all using a single generated project